src/box/lua/schema.lua

```diff
@@ -1677,7 +1677,7 @@ base_index_mt.fselect = function(index, key, opts, fselect_opts)
     local json = require('json')
     for _,t in index:pairs(key, opts) do
         local row = { }
-        for _,f in t:pairs() do
+        for _,f in t:pairs{} do
```

Comment:
You create new table each for iteration in critical part. Do you really need this?